pkg/operator/ceph/controller/spec.go

```diff
@@ -653,8 +655,28 @@ func PrivilegedContext(runAsRoot bool) *v1.SecurityContext {
 	return sec
 }
 
-// LogCollectorContainer runs a cron job to rotate logs
+// LogCollectorContainer rotate logs
 func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Container {
+
+	maxLogSize := resource.NewQuantity(500000000, resource.DecimalSI).String()
+	if c.LogCollector.MaxLogSize != nil {
+		size := c.LogCollector.MaxLogSize.Value() / 1000 / 1000
+		if size == 0 {
+			size = 1
+			logger.Info("maxLogSize is 0M setting to minimum of 1M")
```

Comment:
Why do we even need to set it if it's not present? So far, we haven't needed to use it, so why add it explicitly if not asked by the user? Why do we make it mandatory now?

Reply:
if `MaxLogSIze` is empty then we default to 500M but if uses is giving let's say `500K` it will be `0M`   so instead of setting to default `500M `1M`  is closer to the user input.

@travisn thoughts/

Reply:
500M just seemed like a reasonable default so the logs don't grow too big. There was a case where logs were seen at 100G in less than a week.

Reply:
With this PR, now if maxLogSize is not set, we don't set a default size limit. This seems reasonable since if they even enable the log rotate feature they are likely going to uncomment the maxLogSize property as well, and the example shows 500M by default if it is uncommented. @leseb sound good?